lib/code.rb

```diff
@@ -0,0 +1,53 @@
+require_relative './commands'
+
+class Code
+  include Commands
+  attr_reader :red_pins, :white_pins
+
+  def generate_code
+    @colors = colors = ["r", "g", "o", "y", "b", "p"]
+    @answer = (1..4).map { @colors[rand(6)] }
+  end
+
+  def red_counter(key, input)
+    red_pins = []
+    position = 0
+    while input.length > position
+      key.each do |match|
+        red_pins << 1 if key[position] == input[position]
+        position += 1
+      end
+    end
+    red_pins
+  end
+
+  def trim(key, input)
+   4.times do |n|
+    if input[n] == key[n]
+      input[n] = ""
+      key[n] = ""
+    end
+  end
+  input
+  end
```

Comment:
Can we use map here instead? `4.times` seems pretty unstable to me. What if we change our game to have more characters for input? Let's do this dynamically with an Enumerable (i.e. map)